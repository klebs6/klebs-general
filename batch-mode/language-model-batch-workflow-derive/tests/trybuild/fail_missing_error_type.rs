// ---------------- [ File: tests/trybuild/fail_missing_error_type.rs ]
//
// If the user does not provide `#[batch_error_type(...)]`, parse logic also fails.

use language_model_batch_workflow_derive::LanguageModelBatchWorkflow;

// No `#[batch_error_type(MyErr)]` => fails parse.
#[derive(LanguageModelBatchWorkflow)]
struct MissingErrorType {
    #[batch_client]
    client: (),

    #[batch_workspace]
    workspace: (),

    #[model_type]
    mt: (),
}

// We intentionally do NOT define MyErr or annotate the struct with `#[batch_error_type(MyErr)]`.
// The macro should fail with “Missing required `#[batch_error_type(...)]`.”
fn main() {}
