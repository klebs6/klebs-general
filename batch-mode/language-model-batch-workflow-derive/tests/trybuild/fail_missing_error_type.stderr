error: Missing required `#[system_message]` field.
  --> tests/trybuild/fail_missing_error_type.rs:10:8
   |
10 | struct MissingErrorType {
   |        ^^^^^^^^^^^^^^^^

warning: unused import: `batch_mode_batch_workflow::*`
 --> tests/trybuild/fail_missing_error_type.rs:6:5
  |
6 | use batch_mode_batch_workflow::*;
  |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  |
  = note: `#[warn(unused_imports)]` on by default
